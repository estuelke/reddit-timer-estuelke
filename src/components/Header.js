import React from 'react';
import styled, { css } from 'styled-components';


const HeaderWrapper = styled.header`
  
`;

const Header = styled.header``
  <>
    <div>Logo</div>
    <div>Search</div>
    <div>How it works</div>
    <div>About</div>
  </>
);

export default Header;
