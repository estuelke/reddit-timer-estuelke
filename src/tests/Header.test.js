import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter as Router } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import Header from '../components/Header';
import mainTheme from '../styles/themes';

const renderHeader = () => (
  render(
    <ThemeProvider theme={mainTheme}>
      <Router>
        <Header theme={mainTheme} />
      </Router>
    </ThemeProvider>,
  ));

describe('header', () => {
  it('contains a logo that links to home page', () => {
    const { getByTitle, getByText } = renderHeader();
    const logoElement = getByText('logo.svg');
    const logoLinkElement = getByTitle('header-logo');

    expect(logoElement).toBeInTheDocument();
    expect(logoLinkElement).toContainHTML('href="/"');
  });

  it('contains a link to the search page', () => {
    const { getByText } = renderHeader();
    const searchElement = getByText(/^Search$/);

    expect(searchElement).toBeInTheDocument();
    expect(searchElement).toContainHTML('href="/search"');
  });

  it('contains a link to the about section', () => {
    const { getByText } = renderHeader();
    const aboutElement = getByText(/^About$/);

    expect(aboutElement).toBeInTheDocument();
    expect(aboutElement).toContainHTML('href="/#about"');
  });

  it('contains a link to the How it works section', () => {
    const { getByText } = renderHeader();
    const howItWorksElement = getByText(/^How it works$/);

    expect(howItWorksElement).toBeInTheDocument();
    expect(howItWorksElement).toContainHTML('href="/#how-it-works"');
  });
});
