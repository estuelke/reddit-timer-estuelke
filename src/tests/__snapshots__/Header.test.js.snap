// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="sc-AxirZ gcDtfT"
>
  <a
    href="/"
  >
    <div>
      <svg>
        logo.svg
      </svg>
    </div>
  </a>
  <nav
    className="sc-AxiKw ePdCye"
  >
    <a
      className="sc-AxjAm eKgzTh"
      href="/search?subreddit=javascript"
    >
      Search
    </a>
    <a
      className="sc-AxjAm eKgzTh"
      href="#how-it-works"
    >
      How it works
    </a>
    <a
      className="sc-AxjAm eKgzTh"
      href="#about"
    >
      About
    </a>
  </nav>
</header>
`;
